packages/p2p/src/utils/__tests__/string.spec.ts

```diff
@@ -0,0 +1,18 @@
+import { countDecimalPlaces, getShortNickname } from '../string';
+
+describe('countDecimalPlaces', () => {
+    it('should return the correct number of decimal places', () => {
+        expect(countDecimalPlaces(1.234)).toEqual(3);
+        expect(countDecimalPlaces(1.2)).toEqual(1);
+        expect(countDecimalPlaces(1)).toEqual(0);
+    });
+});
+
```

Comment:
didn't add test case for `generateHexColourFromNickname` as this will be removed soon.